### ISSUE TITLE
Clarinet check does not finish

### ISSUE THREAD

**Issue by friedger** [hirosystems/clarinet#980]
When I add this requirement to clarinet toml file, clarinet check starts and never returns
```
[[project.requirements]]
contract_id = "SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.swap-helper-v1-03"
 ```

Code is here: https://github.com/friedger/clarity-stacking-pools/tree/feat/reward-distributions
using clarinet 1.5.4

**Comment by BowTiedDeployer** [comment]
Same error, same required smart contract `SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.swap-helper-v1-03`
using clarinet 1.5.4

It was running on previous clarinet version https://github.com/stacks-degens/decentralized-mining-pool/blob/main/smart-contracts/utils/exchange-stx-to-btc/contracts/alex-go-swap.clar

**Comment by friedger** [comment]
It looks like clarinet can't handle 
"SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.age000-governance-token"

because it references a contract that does not exist

**Comment by friedger** [comment]
Looks like some circular references:
.fixed-weight-pool
references .alex-vault
references .fixed-weight-pool

**Comment by hugocaillard** [comment]
Yes exactly. I think we broke it when introduced dependency ordering by epoch. Looking into it

**Comment by hugocaillard** [comment]
Actual fix here: #997 

The root cause if this issue is that the [`trait-semi-fungible`](https://explorer.hiro.so/txid/0x74db763fbaa66da3368e642ddac48dc5ac81f3c6e5a9b1aaf358c5745608fdde?chain=mainnet) contains a duplicated functions. It was allowed in Clarity 1 but not anymore in Clarity 2.
We recently changed the default clarity-version value in Clarinet, which was used in the dependency detector (so the trait was wrongly parsed as a Clarity 2 contract)

**Comment by hugocaillard** [comment]
Fixed in #997 and released in 1.6.0.

Sorry it took some time. Thanks for reporting this issue @friedger @BowTiedDeployer
